storybookjs/storybook
# Core: Move `prepareContext` into `store.getStoryContext()`

This way any consumer of the context will get the properly mapped args.

Closes #22099

## What I did

The issue was that although the source decorator emitted the mapped/conditional args, references to `store.getStoryContext().args` would return the unmapped args.

This is due to https://github.com/storybookjs/storybook/pull/20755 which moved the mapping to *before* the decorator chain. 

This PR moves the mapping slightly earlier, so the `context.args` is mapped everywhere (not just inside decorators/render). That will make them inconsistent with `context.initialArgs`, but I think that's probably OK? I'm not sure, this might be considered to risky a change.

In which case we could consider just calling `prepareContext` in the `Source` block.

FTR, this inconsistency between `context.args` + `context.initialArgs` is already the case *inside* the decorator stack/render function, so this would only affect outside callers of `store.getStoryContext()`, which I'm not sure there are any.

## How to test

1. Run react sandbox
2. Add this file:

```tsx
import { Meta, StoryObj } from '@storybook/react';
import React from 'react';

const Example = (props: any) => {
  return <div>Message: {props.hasMessage ? props.message : 'no text'}</div>;
};

type Component = typeof Example;

const meta: Meta<Component> = {
  component: Example,
  tags: ['autodocs'],
  argTypes: {
    hasMessage: { type: 'boolean' },
    message: {
      control: 'text',
      if: { arg: 'hasMessage' },
    },
  },
};

export default meta;

export const Default: StoryObj<Component> = {
  args: {
    hasMessage: false,
    message: 'foo',
  },
};
```

3. Check the source:
<img width="484" alt="image" src="https://user-images.githubusercontent.com/132554/232681971-5ae4533d-1a77-459b-8796-1ad25ab6fbc2.png">


## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
